### ISSUE TITLE
KotlinGenerator should generate implicit joins path accessors as properties

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#13516]
The `KotlinGenerator` could generate additional properties for implicit join paths, such as e.g.:

```kotlin
    /**
     * Get the implicit join path to the <code>public.address</code> table.
     */
    fun address(): Address {
        if (!this::_address.isInitialized)
            _address = Address(this, CUSTOMER__CUSTOMER_ADDRESS_ID_FKEY)

        return _address;
    }

    val address = address()
```

This would allow for this:

```kotlin
CUSTOMER.address.city.country.COUNTRY_
```

Instead of this:

```kotlin
CUSTOMER.address().city().country().COUNTRY_
```

As always, a flag will allow for opting out of this new feature, in case we produce compilation errors, etc. The flag is `<implicitJoinPathsAsKotlinProperties/>`

**Comment by lukaseder** [comment]
In fact, we must implement a `get()` method rather than eagerly assigning the property value:

```kotlin
    val address: Address
        get(): Address = address()
```

**Comment by lukaseder** [comment]
Note, this isn't a breaking change. It seems we can have both styles next to each other.

**Comment by lukaseder** [comment]
Implemented for jOOQ 3.17.0
